src/sentry/api/endpoints/relay_projectids.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.api.permissions import RelayPermission
+from sentry.api.authentication import RelayAuthentication
+from sentry.models import Project, ProjectKey
+
+
+class RelayProjectIdsEndpoint(Endpoint):
+    authentication_classes = (RelayAuthentication,)
+    permission_classes = (RelayPermission,)
+
+    def post(self, request):
+        relay = request.relay
+        assert relay is not None  # should be provided during Authentication
+
+        project_ids = {}
+        for public_key in request.relay_request_data.get("publicKeys") or ():
+            if not ProjectKey.looks_like_api_key(public_key):
+                continue
+
+            try:
+                pk = ProjectKey.objects.get_from_cache(public_key=public_key)
+            except ProjectKey.DoesNotExist:
+                continue
+
+            # NB: Do not validate pk here (is_active or store). Relay should
+            # also receive a mapping for disabled public keys and then perform
+            # the full project config fetch.
+
+            project = Project.objects.get_from_cache(id=pk.project_id)
```

Comment:
We already leak out that information from the loader endpoint so we do not consider it confidential. We should skip the check here as a result.